clang/lib/StaticAnalyzer/Checkers/FixedAddressChecker.cpp

```diff
@@ -63,11 +62,39 @@ void FixedAddressChecker::checkPreStmt(const BinaryOperator *B,
         "Using a fixed address is not portable because that address will "
         "probably not be valid in all environments or platforms.";
     auto R = std::make_unique<PathSensitiveBugReport>(BT, Msg, N);
-    R->addRange(B->getRHS()->getSourceRange());
+    R->addRange(SrcExpr->getSourceRange());
     C.emitReport(std::move(R));
   }
 }
 
+void FixedAddressChecker::checkPreStmt(const BinaryOperator *B,
+                                       CheckerContext &C) const {
+  if (B->getOpcode() != BO_Assign)
+    return;
+
+  checkUseOfFixedAddress(B->getType(), B->getRHS(), C);
+}
+
+void FixedAddressChecker::checkPreStmt(const DeclStmt *D,
+                                       CheckerContext &C) const {
+  for (const auto *D1 : D->decls()) {
```

Comment:
I suppose you use D1 name here because D is already taken. Maybe you could use "Ds" instead of "D" to free it up to be used here.

Reply:
I agree that "`DS`" / "`D`" would be a more traditional naming choice, but actually there's something poetic in abbreviating  "one declaration from `D`" as  "`D1`".

Reply:
But "D" is already singular. And eventually we reach the second, thirds, etc decls too. In which case number 1 would confuse me, because its not only the first one.